or_benchmarks/src/CheckerResultModule.cpp

```diff
@@ -0,0 +1,243 @@
+    }
+
+    else if (methodname == "CheckCollision_link_with_exclusions")
+    {
+        OpenRAVE::KinBody::LinkConstPtr plink = DeserializeLink("link",check_log_val);
+
+        std::vector<OpenRAVE::KinBodyConstPtr> vbodiesexcluded = DeserializeKinBodyList("bodies_excluded",check_log_val);
+        std::vector<OpenRAVE::KinBody::LinkConstPtr> vlinksexcluded = DeserializeLinkList("links_excluded",check_log_val);
+
+        start = std::chrono::high_resolution_clock::now();
+        check_result = checker_ptr -> CheckCollision(plink, vbodiesexcluded, vlinksexcluded);
+        end = std::chrono::high_resolution_clock::now();
+    }
+
+    else if (methodname == "CheckCollision_body_with_exclusions")
+    {
+        OpenRAVE::KinBodyConstPtr pbody = DeserializeKinBody("body",check_log_val);
+
+        std::vector<OpenRAVE::KinBodyConstPtr> vbodiesexcluded = DeserializeKinBodyList("bodies_excluded",check_log_val);
+        std::vector<OpenRAVE::KinBody::LinkConstPtr> vlinksexcluded = DeserializeLinkList("links_excluded",check_log_val);
+
+        start = std::chrono::high_resolution_clock::now();
+        check_result = checker_ptr -> CheckCollision(pbody, vbodiesexcluded, vlinksexcluded);
+        end = std::chrono::high_resolution_clock::now();
+    }
+
+    else if (methodname == "CheckStandaloneSelfCollision")
+    {
+        OpenRAVE::KinBodyConstPtr pbody = DeserializeKinBody("body",check_log_val);
+
+        start = std::chrono::high_resolution_clock::now();
+        check_result = checker_ptr -> CheckStandaloneSelfCollision(pbody);
+        end = std::chrono::high_resolution_clock::now();
+    }
+
+    else if (methodname == "CheckStandaloneSelfCollisionBaked")
+    {
+        OpenRAVE::KinBodyConstPtr pbody = DeserializeKinBody("body",check_log_val);
+
+        start = std::chrono::high_resolution_clock::now();
+        check_result = checker_ptr -> CheckStandaloneSelfCollision(pbody);
+        end = std::chrono::high_resolution_clock::now();
+    }
+
+    else
+    {
+        throw std::runtime_error("Unknown method - "+methodname);
+    }
+
+    std::chrono::duration<double> elapsed_seconds = end-start;
+    time_result = elapsed_seconds.count();
+
+    sout<<check_result<<" "<<time_result;
+
+
+    return true;
+
+}
+
+OpenRAVE::KinBodyConstPtr CheckerResultModule::DeserializeKinBody(std::string keyname, picojson::value check_log_val)
+{
+    std::string body_name(check_log_val.get(keyname).get<std::string>());
+    OpenRAVE::KinBodyConstPtr pbody = GetEnv() -> GetKinBody(body_name);
+    return pbody;
+}
+
+OpenRAVE::KinBody::LinkConstPtr CheckerResultModule::DeserializeLink(std::string keyname, picojson::value check_log_val)
+{
+    std::string link_and_body(check_log_val.get(keyname).get<std::string>());
+
+    size_t first_blank_pos = link_and_body.find(' ');
+
+    std::string link_name = link_and_body.substr(0,first_blank_pos);
+    std::string body_name = link_and_body.substr(first_blank_pos+1);
```

Comment:
It would be more elegant to use a JSON object with separate `link_name` and `body_name` keys. 😞 Would doing this break compatibility with the data that you already have collected?